src/authorizations/utils/permissions.ts

```diff
@@ -135,3 +135,72 @@ export enum BucketTab {
   AllBuckets = 'All Buckets',
   Scoped = 'Scoped',
 }
+
+export const formatPermissionsObj = permissions => {
+  const newPerms = permissions.reduce((acc, {action, resource}) => {
+    const {type, id, orgID, name} = resource
+    let p
+
+    if (acc.hasOwnProperty(type)) {
+      p = {...acc[type]}
+      if (id && (type === 'buckets' || type === 'telegrafs')) {
+        if (p.sublevelPermissions.hasOwnProperty(id)) {
+          p.sublevelPermissions[id].permissions[action] = true
+        } else {
+          p.sublevelPermissions[id] = {
+            id: id,
+            orgID: orgID,
+            name: name,
+            permissions: {
+              read: action === 'read',
+              write: action === 'write',
+            },
+          }
+        }
+      } else {
+        p[action] = true
+      }
+    } else {
+      if (id && (type === 'buckets' || type === 'telegrafs')) {
+        p = {
+          read: false,
+          write: false,
+          sublevelPermissions: {
+            [id]: {
+              id: id,
+              orgID: orgID,
+              name: name,
+              permissions: {
+                read: action === 'read',
+                write: action === 'write',
+              },
+            },
+          },
+        }
+      } else {
+        p = {
+          read: action === 'read',
+          write: action === 'write',
+        }
+      }
+    }
+
+    return {...acc, [type]: p}
+  }, {})
+
+  Object.keys(newPerms).map(resource => {
```

Comment:
```suggestion
  Object.keys(newPerms).forEach(resource => {
```

`map` and `forEach` both loop over every item of an array and apply the specified function to the array. The difference is that `map` produces an array of the same size as the input array and expects the function to return the new item of the array. `forEach` simply iterates over the array and does some operation. `forEach` is designed to be used with side effects like you're doing here.